conda_forge_tick/migrators/cross_compile.py

```diff
@@ -347,10 +351,20 @@ def migrate(self, recipe_dir: str, attrs: "AttrsTypedDict", **kwargs: Any) -> No
             for line in meta_yaml:
                 if previous_was_build:
                     nspaces = len(line) - len(line.lstrip())
-                    new_lines.append(
-                        " " * nspaces
-                        + "- cross-r-base {{ r_base }}  # [build_platform != target_platform]\n",
-                    )
+                    if recipe_file == "recipe.yaml":
+                        new_lines.extend(
+                            [
+                                " " * nspaces
+                                + "- if: build_platform != target_platform\n",
+                                " " * nspaces + "  then:\n",
+                                " " * nspaces + "    - cross-r-base ${{ r_base }}\n",
+                            ]
```

Comment:
FWICS we could just append all the lines as a single element here, but I've figured out that since it's a list of lines by design, it's cleaner to add them separately.